plugins/woocommerce/client/legacy/js/frontend/order-attribution.js

```diff
@@ -111,4 +114,28 @@
 		};
 	}
 
+	// Work around the lack of explicit script dependency for the checkout block.
+	// Conditionally, wait for and use 'wp-data' & 'wc-blocks-checkout.
+
+	// Wait for (async) block checkout initialization and set source values once loaded.
+	function eventuallyInitializeCheckoutBlock() {
+		if (
+			window.wp && window.wp.data && typeof window.wp.data.subscribe === 'function'
+		) {
+			const unsubscribe = wp.data.subscribe( function () {
+				const checkoutDataStore = wp.data.select( CHECKOUT_STORE_KEY );
+				if ( undefined !== checkoutDataStore && checkoutDataStore.getOrderId() !== 0 ) {
```

Comment:
```suggestion
				if ( 'undefined' !== typeof checkoutDataStore && checkoutDataStore.getOrderId() !== 0 ) {
```

I prefer this way of doing it as it's a little safer.

Reply:
`checkoutDataStore` constant is declared just the line above, so IMHO typeof is not needed.

Also, do we even need that check? We subscribed to `CHECKOUT_STORE_KEY` store changes, so `CHECKOUT_STORE_KEY` store should be guaranteed to exist in the callback, shouldn't it?

Reply:
I did a quick micro-perf check, and it seems there is no significant difference between them. https://jsperf.app/nofavo

Using `type of` here would make lines 127 and below "safe" in the sense they will run without visible error when line 126 is lost (for example, in a bad merge).
IMHO, this is actually **not desired**. If the line 126 is lost, the code should fail fast. So we could identify the bad merge or other bug before shipping it to production.

Reply:
I removed this condition altogether in https://github.com/woocommerce/woocommerce/pull/43857/commits/f3abaf0304f75d5a0edf2a4e7709cd344ef4bfd3

We don't need to check if there is an order being processed in the checkout.
To add extension data, we only need the store to finally be there. That should be guaranteed by the  `subscribe` itself.

Reply:
@tomalec I appreciate how deeply you considered this! Kudos. 🙏🏼 The result is better for sure.